src/main/java/com/atlassian/db/replica/api/ThrottledCache.java

```diff
@@ -0,0 +1,35 @@
+package com.atlassian.db.replica.api;
+
+import com.atlassian.db.replica.internal.util.ThreadSafe;
+import com.atlassian.db.replica.spi.SuppliedCache;
+
+import java.util.Optional;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Supplier;
+
+
+/**
+ * Every read can potentially refresh the cache, however it can be loaded only by one thread at the time.
+ * Other threads instead of waiting are getting last known value.
+ */
+@ThreadSafe
+public class ThrottledCache<T> implements SuppliedCache<T> {
+    private final ReentrantLock lock = new ReentrantLock();
+    private T value = null;
```

Comment:
Assigning null here is not necessary. If your intention was to emphasise it then it can stay ;)

Reply:
Implicit defaults are easier to read